src/main/java/com/datapipe/jenkins/vault/credentials/VaultKubernetesCredential.java

```diff
@@ -0,0 +1,63 @@
+package com.datapipe.jenkins.vault.credentials;
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultException;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.datapipe.jenkins.vault.exception.VaultPluginException;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Collectors;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class VaultKubernetesCredential extends AbstractVaultTokenCredential {
+
+    private static final String SERVICE_ACCOUNT_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
```

Comment:
This path is on the same as the one on Windows?

Reply:
I don't really know that. The windows container support in kubernetes is relatively new and after a quick search I couldn't find any documentation on how the volumes work in windows containers, but I guess as in any regular windows.
The service account token is just a regular secret, so technically it can be mounted on any path even in unix containers. But the Service Account Admission Controller, which manages service accounts in kubernetes, mounts the token on this path by default. [link to the documentation](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-admission-controller)

Reply:
Perhaps this option should only be visible if kubernetes secret is detected?

Reply:
Can be achieved with `isApplicable` in the descriptor

Reply:
> Perhaps this option should only be visible if kubernetes secret is detected?

Do you mean to only show the VaultKubernetesCredential as option, when the mounted token is detected?

Reply:
Simply a suggestion, don't know if it makes sense. I am fine with the feature as is.